### ISSUE TITLE
`Once::call_once` doesn't guarantee that the given closure is called only if this is the first time `call_once` has been called.

### ISSUE THREAD

**Issue by usamoi** [mvdnes/spin-rs#117]
This test fails with a certain probability.

```rust
// insert this in once.rs:684
    #[test]
    fn extra() {
        use std::sync::Arc;
        use std::sync::atomic::AtomicUsize;
        use std::time::Duration;
        let share = Arc::new(AtomicUsize::new(0));
        let once = Arc::new(Once::<_, Spin>::new());
        let mut hs = Vec::new();
        for _ in 0..10 {
            let h = thread::spawn({
                let share = share.clone();
                let once = once.clone();
                move || {
                    thread::sleep(Duration::from_millis(10));
                    once.call_once(|| {
                        share.fetch_add(1, Ordering::SeqCst);
                    });
                }
            });
            hs.push(h);
        }
        for h in hs {
            let _ = h.join();
        }
        assert_eq!(1, share.load(Ordering::SeqCst));
    }
```

**Comment by usamoi** [comment]
A typo in https://github.com/mvdnes/spin-rs/blob/master/src/once.rs#L113 makes this bug.
I have made a pull request (https://github.com/mvdnes/spin-rs/pull/118) to fix it.

**Comment by zesterer** [comment]
This has now been included in the `0.9.3` release.
